toolchain/sem_ir/id_kind.h

```diff
@@ -115,15 +115,21 @@ class TypeEnum {
 };
 
 // An enum of all the ID types used as instruction operands.
+//
+// As instruction operands, the types listed here can appear as fields of typed
+// instructions (`toolchain/sem_ir/typed_insts.h`) and must implement the
+// `FromRaw` and `ToRaw` protocol in `SemIR::Inst`. In most cases this is done
+// by inheriting from `IdBase` or `IndexBase`.
 using IdKind = TypeEnum<
     // From base/value_store.h.
     IntId, RealId, FloatId, StringLiteralValueId,
-    // From sem_ir/id.h.
-    InstId, AbsoluteInstId, ConstantId, EntityNameId, CompileTimeBindIndex,
-    RuntimeParamIndex, FacetTypeId, FunctionId, ClassId, InterfaceId, ImplId,
-    GenericId, SpecificId, ImportIRId, ImportIRInstId, LocId, BoolValue,
-    IntKind, NameId, NameScopeId, InstBlockId, StructTypeFieldsId, TypeId,
-    TypeBlockId, ElementIndex, LibraryNameId, FloatKind>;
+    // From sem_ir/ids.h.
+    InstId, AbsoluteInstId, AnyRawId, ConstantId, EntityNameId,
```

Comment:
Hm, on reflection, do we want to list `AnyRawId` here? I think it could be reasonable to say that we don't allow it as the type of an instruction operand, only as the type of an instruction category operand.

Reply:
Ah, I see, you have a matching requirement on `FromRaw`. Let's leave this alone then; I don't think we're likely to add `AnyRawId` fields to instructions by accident :)

Reply:
I think that's right, yeah, it was my intention for it as well, and the reason for the `Any` prefix. If we remove it from here, we'd have to add special cases to the documentation and an overload for it to FromRaw (it requires IdKind contains the type) but not to ToRaw (since it's an IdBase now). That's a bit of an asymetry. If it's not an IdKind then I'd probably prefer to not make it an IdBase either, so two overloads.

Reply:
Ok SGTM